### ISSUE TITLE
support only one protocol？

### ISSUE THREAD

**Issue by LesterSixNet** [shamblett/mqtt_client#138]
![image](https://user-images.githubusercontent.com/47265717/69514806-8683af00-0f87-11ea-8520-c02321ebe830.png)

**Comment by shamblett** [comment]
No, you can change them or not use them at all, see the websocketProtocols setter on the client

**Comment by LesterSixNet** [comment]
![image](https://user-images.githubusercontent.com/47265717/69529291-f0fa1680-0faa-11ea-864a-10bc0fd1ce2c.png)

![image](https://user-images.githubusercontent.com/47265717/69529249-da53bf80-0faa-11ea-9194-3ac1ea4a281f.png)

I dont't set a string "mqtt"

**Comment by LesterSixNet** [comment]
Server Only support protocol "mqtt", not "[mqtt]"

**Comment by shamblett** [comment]
The protocols are just a list of strings that you pass to the dart websocket connect call, 
```
static Future<WebSocket> connect(String url,
          {Iterable<String> protocols,
          Map<String, dynamic> headers,
          CompressionOptions compression:
              CompressionOptions.compressionDefault}) =>
      _WebSocketImpl.connect(url, protocols, headers, compression: compression);
```
as far as I know this should result in 'mqtt' being used, not [mqtt] as you are seeing.

**Comment by LesterSixNet** [comment]
thanks，I know.

**Comment by LesterSixNet** [comment]
![image](https://user-images.githubusercontent.com/47265717/69533743-59e58c80-0fb3-11ea-9ac9-082cf4fc2c34.png)
I try change **$protocols** to  **mqtt**, connected success.

**Comment by shamblett** [comment]
I don't know what you mean by '$protocols', what exactly do you think is wrong in the client?

**Comment by LesterSixNet** [comment]
![image](https://user-images.githubusercontent.com/47265717/69534724-581cc880-0fb5-11ea-991d-6d863f51d186.png)

protocols is List, toString "[mqtt]"

**Comment by LesterSixNet** [comment]
![image](https://user-images.githubusercontent.com/47265717/69535149-33752080-0fb6-11ea-9262-0bc00e56d3b0.png)
![image](https://user-images.githubusercontent.com/47265717/69535105-1ccec980-0fb6-11ea-84c6-0b6dcd94d9a1.png)


sorry, server configuration support only mqtt protocol upgrade

**Comment by LesterSixNet** [comment]
![image](https://user-images.githubusercontent.com/47265717/69535238-76cf8f00-0fb6-11ea-9277-9b9f675e22be.png)
![image](https://user-images.githubusercontent.com/47265717/69535223-6ddebd80-0fb6-11ea-9792-794078128e60.png)

sorry, server configuration only support mqtt upgrade protocol

**Comment by LesterSixNet** [comment]
sorry, server current configuration only support mqtt upgrade protocol

**Comment by shamblett** [comment]
Yes, I see it now in the ws2connection handler, not the ws one.  I'll fix this to use the string without the '[]'

**Comment by shamblett** [comment]
Fix applied, package re-released at 5.6.2
